### PR TITLE
make rocksdb statistics configurable

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
When debugging performance, we can get more details of rocksdb by enable rocksdb's statistics. 

```
** Level 0 read latency histogram (micros):
Count: 7384 Average: 8.5678  StdDev: 6.53
Min: 0  Median: 7.0683  Max: 212
Percentiles: P50: 7.07 P75: 8.53 P99: 19.55 P99.9: 132.32 P99.99: 212.00
------------------------------------------------------
[       0,       1 )       20   0.271%   0.271%
[       1,       2 )        6   0.081%   0.352%
[       2,       3 )      162   2.194%   2.546%
[       3,       4 )      501   6.785%   9.331% #
[       4,       5 )      294   3.982%  13.313% #
[       5,       6 )     1036  14.030%  27.343% ###
[       6,       7 )     1569  21.249%  48.592% ####
[       7,       8 )     1523  20.626%  69.217% ####
[       8,       9 )      807  10.929%  80.146% ##
[       9,      10 )      311   4.212%  84.358% #
[      10,      12 )      142   1.923%  86.281%
[      12,      14 )      177   2.397%  88.678%
[      14,      16 )      433   5.864%  94.542% #
[      16,      18 )      265   3.589%  98.131% #
[      18,      20 )       83   1.124%  99.255%
[      20,      25 )       29   0.393%  99.648%
[      25,      30 )        9   0.122%  99.770%
[      30,      35 )        2   0.027%  99.797%
[      35,      40 )        4   0.054%  99.851%
[      40,      45 )        2   0.027%  99.878%
[      70,      80 )        1   0.014%  99.892%
[     120,     140 )        1   0.014%  99.905%
[     140,     160 )        2   0.027%  99.932%
[     160,     180 )        3   0.041%  99.973%
[     180,     200 )        1   0.014%  99.986%
[     200,     250 )        1   0.014% 100.000%
...
rocksdb.block.cache.hit COUNT : 9158202
rocksdb.block.cache.add COUNT : 36531
rocksdb.block.cache.add.failures COUNT : 0
rocksdb.block.cache.index.miss COUNT : 0
rocksdb.block.cache.index.hit COUNT : 0
rocksdb.block.cache.index.bytes.insert COUNT : 0
rocksdb.block.cache.index.bytes.evict COUNT : 0
rocksdb.block.cache.filter.miss COUNT : 0
rocksdb.block.cache.filter.hit COUNT : 0
rocksdb.block.cache.filter.bytes.insert COUNT : 0
rocksdb.block.cache.filter.bytes.evict COUNT : 0
rocksdb.block.cache.data.miss COUNT : 315662
rocksdb.block.cache.data.hit COUNT : 9158203
rocksdb.block.cache.bytes.read COUNT : 122705707392
rocksdb.block.cache.bytes.write COUNT : 2005227328
rocksdb.bloom.filter.useful COUNT : 50149930
rocksdb.persistent.cache.hit COUNT : 0
rocksdb.persistent.cache.miss COUNT : 0
rocksdb.sim.block.cache.hit COUNT : 0
rocksdb.sim.block.cache.miss COUNT : 0
rocksdb.memtable.hit COUNT : 7518202
rocksdb.memtable.miss COUNT : 110424648
...
```

### CODE REVIEW

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -194,6 +194,21 @@ fn get_rocksdb_option(matches: &Matches, config: &toml::Value) -> RocksdbOptions
     let wal_recovery_mode = util::config::parse_rocksdb_wal_recovery_mode(rmode).unwrap();
     opts.set_wal_recovery_mode(wal_recovery_mode);
 
+    let enable_statistics = config.lookup("rocksdb.enable-statistics")
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: why use config lookup here? I think we should support get_bool_value here. 
/cc @overvenus

**Comment by siddontang** [comment]
LGTM

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -194,6 +218,23 @@ fn get_rocksdb_option(matches: &Matches, config: &toml::Value) -> RocksdbOptions
...
+    let enable_statistics = get_boolean_value("",
+                                              "rocksdb.enable-statistics",
+                                              matches,
+                                              config,
+                                              Some(false),
+                                              |v| v.as_bool());
+    if enable_statistics {
+        opts.enable_statistics();
+    }
+    let stats_dump_period_sec = get_integer_value("",
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: This operation only needed when `enable_statistics` is true.

Comment (zhangjinpeng1987) [inline src/bin/tikv-server.rs]: If enalbe_statistics is false, rocksdb also dump some statistics, if enalbe_statistics is true the statistics include more detail informations.

**Pull request merged**